libminifi/src/FlowController.cpp

```diff
@@ -320,12 +320,12 @@ void FlowController::initializeUninitializedSchedulers() {
   conditionalReloadScheduler<CronDrivenSchedulingAgent>(cron_scheduler_, !cron_scheduler_);
 }
 
-void FlowController::reinitializeSchedulersWithNewThreadPool() {
-  using ControllerServiceProvider = core::controller::ControllerServiceProvider;
+void FlowController::reinitializeSchedulersWithClearedThreadPool() {
+  using NonNullControllerServiceProviderPtr = gsl::not_null<core::controller::ControllerServiceProvider*>;
   restartThreadPool();
-  timer_scheduler_ = std::make_shared<TimerDrivenSchedulingAgent>(gsl::not_null<ControllerServiceProvider*>(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
-  event_scheduler_ = std::make_shared<EventDrivenSchedulingAgent>(gsl::not_null<ControllerServiceProvider*>(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
-  cron_scheduler_ = std::make_shared<CronDrivenSchedulingAgent>(gsl::not_null<ControllerServiceProvider*>(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
+  timer_scheduler_ = std::make_shared<TimerDrivenSchedulingAgent>(NonNullControllerServiceProviderPtr(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
+  event_scheduler_ = std::make_shared<EventDrivenSchedulingAgent>(NonNullControllerServiceProviderPtr(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
+  cron_scheduler_ = std::make_shared<CronDrivenSchedulingAgent>(NonNullControllerServiceProviderPtr(this), provenance_repo_, flow_file_repo_, content_repo_, configuration_, thread_pool_);
```

Comment:
Consider using `gsl::make_not_null` and relying on implicit conversion to base class pointer.

Reply:
This is an explicit conversion.

Reply:
What is the advantake of using `make_not_null`? It does nothing, does it?